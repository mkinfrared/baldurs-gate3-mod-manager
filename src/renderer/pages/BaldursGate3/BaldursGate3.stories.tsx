import { Meta, StoryObj } from "@storybook/react";

import { BaldursGate3 } from "./BaldursGate3";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/BaldursGate3",
  component: BaldursGate3,
  tags: ["autodocs"],
} satisfies Meta<typeof BaldursGate3>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <BaldursGate3 {...args} />
    </div>
  ),
};

export { Default };

export default meta;
