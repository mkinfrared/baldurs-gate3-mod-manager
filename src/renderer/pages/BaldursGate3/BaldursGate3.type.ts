export type BaldursGate3Props = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};
