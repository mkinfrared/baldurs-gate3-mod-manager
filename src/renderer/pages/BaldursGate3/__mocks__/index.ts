export { default } from "./BaldursGate3.mock";
