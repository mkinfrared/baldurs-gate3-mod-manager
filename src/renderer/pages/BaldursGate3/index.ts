export { BaldursGate3 } from "./BaldursGate3";

export * from "./BaldursGate3.type";
