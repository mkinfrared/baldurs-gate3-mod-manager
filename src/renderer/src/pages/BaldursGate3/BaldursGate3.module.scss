.BaldursGate3 {
  display: grid;
  gap: var(--spacing-10);
  grid-template-columns: repeat(auto-fit, minmax(350px, 1fr));
  min-height: 100dvh;
}
