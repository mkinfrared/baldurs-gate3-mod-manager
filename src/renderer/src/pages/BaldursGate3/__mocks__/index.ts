export { BaldursGate3Mock as BaldursGate3 } from "./BaldursGate3.mock";
