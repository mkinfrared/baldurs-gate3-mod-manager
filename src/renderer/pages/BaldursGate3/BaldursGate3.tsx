import { classNames } from "renderer/shared/lib/helpers";
import { InstalledModsSection, ModFileSection } from "renderer/widgets/ui";

import css from "./BaldursGate3.module.scss";
import { BaldursGate3Props } from "./BaldursGate3.type";

const BaldursGate3 = ({ className }: BaldursGate3Props) => (
  <div
    className={classNames(css.BaldursGate3, className)}
    data-testid="BaldursGate3"
  >
    <InstalledModsSection />
    <ModFileSection />
  </div>
);

export { BaldursGate3 };
