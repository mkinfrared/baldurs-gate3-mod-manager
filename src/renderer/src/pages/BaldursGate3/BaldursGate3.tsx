import { classNames } from "@renderer/shared/lib/helpers";
import { InstalledModsSection, ModFileSection } from "@renderer/widgets";

import css from "./BaldursGate3.module.scss";
import { BaldursGate3Props } from "./BaldursGate3.type";

const BaldursGate3 = ({ className }: BaldursGate3Props) => {
  const game = "BG3";

  return (
    <div
      className={classNames(css.BaldursGate3, className)}
      data-testid="BaldursGate3"
    >
      <InstalledModsSection game={game} />
      <ModFileSection game={game} />
    </div>
  );
};

export { BaldursGate3 };
