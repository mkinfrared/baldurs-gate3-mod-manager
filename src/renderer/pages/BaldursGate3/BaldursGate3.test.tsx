import { render } from "@testing-library/react";

import { BaldursGate3 } from "./BaldursGate3";

jest.mock("renderer/widgets/ui");

describe("<BaldursGate3 />", () => {
  const Component = <BaldursGate3 />;

  it("should be defined", () => {
    expect(BaldursGate3).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("BaldursGate3");

    expect(element).toBeDefined();
  });
});
