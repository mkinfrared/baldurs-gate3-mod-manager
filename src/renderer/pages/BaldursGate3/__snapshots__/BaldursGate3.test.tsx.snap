// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaldursGate3 /> should match the snapshot 1`] = `
<div>
  <div
    class="BaldursGate3"
    data-testid="BaldursGate3"
  />
</div>
`;
